pkg/updatestrategy/aws_asg.go

```diff
@@ -129,23 +131,128 @@ func (n *ASGNodePoolsBackend) Scale(nodePool *api.NodePool, replicas int) error
+// just removes a tag.
+func (n *ASGNodePoolsBackend) SuspendAutoscaling(nodePool *api.NodePool) error {
+	tags := map[string]string{
+		kubeAutoScalerEnabledTagKey: "",
+	}
+	return n.deleteTags(nodePool, tags)
+}
+
+// deleteTags deletes the specified tags from the node pool asg.
+func (n *ASGNodePoolsBackend) deleteTags(nodePool *api.NodePool, tags map[string]string) error {
+	asg, err := n.getNodePoolASG(nodePool)
+	if err != nil {
+		return err
+	}
+
+	asgTags := make([]*autoscaling.Tag, 0, len(tags))
+
+	for key, val := range tags {
+		tag := &autoscaling.Tag{
+			Key:          aws.String(key),
+			Value:        aws.String(val),
+			ResourceId:   asg.AutoScalingGroupName,
+			ResourceType: aws.String("auto-scaling-group"),
+		}
+		asgTags = append(asgTags, tag)
+	}
+
+	params := &autoscaling.DeleteTagsInput{
+		Tags: asgTags,
+	}
+
+	_, err = n.asgClient.DeleteTags(params)
+	return err
+}
+
 // Terminate terminates a node from the ASG and optionally decrements the
 // DesiredCapacity. By default the desired capacity will not be decremented.
+// In case the new desired capacity is less then the current min size of the
+// ASG, it will also decrease the ASG minSize.
 func (n *ASGNodePoolsBackend) Terminate(node *Node, decrementDesired bool) error {
-	instanceId := aws.String(instanceIDFromProviderID(node.ProviderID, node.FailureDomain))
+	instanceId := instanceIDFromProviderID(node.ProviderID, node.FailureDomain)
+
+	// if desired should be decremented check if we also need to decrement
+	// the minSize of the ASG.
+	if decrementDesired {
+		// lookup ASG name in the EC2 tags of the instance
+		var asgName string
+		params := &ec2.DescribeTagsInput{
+			Filters: []*ec2.Filter{
+				{
+					Name:   aws.String("resource-id"),
+					Values: []*string{aws.String(instanceId)},
+				},
+				{
+					Name:   aws.String("key"),
+					Values: []*string{aws.String(ec2AutoscalingGroupTagKey)},
+				},
+			},
+		}
+		err := n.ec2Client.DescribeTagsPages(params, func(resp *ec2.DescribeTagsOutput, lastPage bool) bool {
+			for _, tag := range resp.Tags {
+				if aws.StringValue(tag.Key) == ec2AutoscalingGroupTagKey {
+					asgName = aws.StringValue(tag.Value)
+					return false
+				}
+			}
+			return true
+		})
+		if err != nil {
+			return err
+		}
+
+		if asgName == "" {
+			return fmt.Errorf("failed to get Autoscaling Group name from EC2 tags of instance '%s'", instanceId)
+		}
+
+		// get current sizes in the ASG
+		asgParams := &autoscaling.DescribeAutoScalingGroupsInput{
+			AutoScalingGroupNames: []*string{aws.String(asgName)},
+		}
+
+		resp, err := n.asgClient.DescribeAutoScalingGroups(asgParams)
+		if err != nil {
+			return err
+		}
+
+		if len(resp.AutoScalingGroups) == 0 {
+			return fmt.Errorf("failed to find ASG '%s'", asgName)
+		}
+
+		asg := resp.AutoScalingGroups[0]
+
+		newDesired := aws.Int64Value(asg.DesiredCapacity) - 1
+		minSize := aws.Int64Value(asg.MinSize)
+
+		if minSize > 0 && newDesired >= 0 && minSize > newDesired {
```

Comment:
nit: I think it would be better readable if you sort these like this, because you have at left most the smalles and rightmost the largest:

    if 0 <= newDesired && 0 < minSize && newDesired < minSize {